openworld/sdk/generator/client/visitors/operations.py

```diff
@@ -11,21 +11,64 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+import collections
+import re
 from pathlib import Path
 
+from datamodel_code_generator.model import DataModel
 from datamodel_code_generator.model.pydantic import CustomRootType
-from fastapi_code_generator.parser import OpenAPIParser, Operation
+from fastapi_code_generator.parser import Argument, OpenAPIParser, Operation
 from fastapi_code_generator.visitor import Visitor
 
 
-def get_operations(parser: OpenAPIParser, model_path: Path) -> dict[str, object]:
-    sorted_operations: list[Operation] = []
+def parse_non_schema_models(parser: OpenAPIParser):
+    models: list[DataModel] = list(
+        {
+            result
+            for result in parser.results
+            if isinstance(result, DataModel)
+            and result.reference.path.startswith("#/paths/")
+            and not result.class_name.endswith("Response")
+            and not result.class_name.endswith("Request")
+        }
+    )
+
+    models_classnames = [model.class_name for model in models]
+    for index in range(len(models) - 1, -1, -1):
+        model = models[index]
+        if model.class_name[-1].isdigit() or f"{model.class_name}1" in models_classnames:
+            continue
+        models.pop(index)
+        index += 1
+
+    return models
+
+
+def clean_unused_models(parser: OpenAPIParser, unused_models_classnames: list[str]):
+    indices_stack = list()
+    for index, result in enumerate(parser.results):
+        if isinstance(result, DataModel) and result.class_name in unused_models_classnames:
+            indices_stack.append(index)
+
+    while len(indices_stack):
+        parser.results.pop(indices_stack.pop())
+
+
+def update_classname_by_operation_id(operation_id: str, classname: str):
+    if not operation_id:
+        return classname
 
-    models = dict()
+    if not classname:
+        classname = ""
 
-    for model in parser.results:
-        models[model.name] = model
+    operation_id = list(operation_id)
+    operation_id[0] = operation_id[0].upper()
```

Comment:
What are we trying to do here? (64-67)

Reply:
Get the `operation_id` (method name in camel case as per specs), capitalize it so we can use it as a prefix to generate a new classname for models (e.g. `Include1` in `get_regions` becomes `GetRegoinsInclude`).

`"String"/capitalize()` method wont work as it will not preserve the camel case for rest of the string.